common/src/chain/transaction/output/timelock.rs

```diff
@@ -0,0 +1,15 @@
+use serialization::{Decode, Encode};
+
+use crate::{chain::block::timestamp::BlockTimestamp, primitives::BlockHeight};
+
+#[derive(Debug, Clone, Ord, PartialOrd, Eq, PartialEq, Encode, Decode)]
+pub enum OutputTimeLock {
+    #[codec(index = 0)]
+    UntilHeight(BlockHeight),
+    #[codec(index = 1)]
+    UntilTime(BlockTimestamp),
+    #[codec(index = 2)]
+    ForBlockCount(#[codec(compact)] u64),
```

Comment:
How about making this the `BlockDistance` type instead of `u64`? Semantically it seems a good match but there might be an issue with negative distances. Maybe `BlockDistance` should be unsigned after all, since it is used in a way where we know which height is supposed to come first.

Reply:
Funny enough, if you look into the history, you'll see that it was BlockDistance. The issue though is that compact encoding isn't supported for signed types. In the big picture, this isn't a concern now. We can beautify this later since there's a technical issue behind it.